python/ray/tests/test_node_label_scheduling_strategy.py

```diff
@@ -205,13 +214,53 @@ def test_node_label_scheduling_with_soft(ray_start_cluster):
         ray.get(actor.get_node_id.remote(), timeout=3)
 
 
-def get_new_node_id(created_nodes):
-    wait_for_condition(lambda: len(ray.nodes()) > len(created_nodes), timeout=30)
-    nodes = ray.nodes()
-    for node in nodes:
-        if node["NodeID"] not in created_nodes:
-            created_nodes.append(node["NodeID"])
-            return node["NodeID"]
+def test_node_not_available(ray_start_cluster):
+    cluster = ray_start_cluster
+    created_nodes = []
+    cluster.add_node(num_cpus=1, labels={"gpu_type": "A100"})
+    cluster.wait_for_nodes()
+    ray.init(address=cluster.address)
+    node_1 = get_new_node_id(created_nodes)
```

Comment:
A better way to get node id is assigning each node a unique resource and run a task to get node id

```
cluster.add_node(resources={"worker1": 1})
ray.get(get_node_id.options(resources={"worker1": 1}).remote())
```

Reply:
Fixed.